Build failure: gotop

### Steps To Reproduce

Steps to reproduce the behavior:
1. nix-env -i gotop

### Build log

```
o: downloading gopkg.in/yaml.v2 v2.4.0
go: downloading github.com/mitchellh/go-homedir v1.1.0
# github.com/shirou/gopsutil/cpu
../go/pkg/mod/github.com/shirou/gopsutil@v3.20.12+incompatible/cpu/cpu_darwin_cgo.go:13:5: error: 'TARGET_OS_MAC' is not defined, evaluates to 0 [-Werror,-Wundef-prefix=TARGET_OS_]
#if TARGET_OS_MAC
    ^
1 error generated.
error: builder for '/nix/store/8iqqlbzyk6mflmmzh8cc89h4aynb7r6g-gotop-4.2.0.drv' failed with exit code 1;
       last 10 log lines:
       > go: downloading github.com/pkg/errors v0.9.1
       > go: downloading github.com/ghodss/yaml v1.0.0
       > go: downloading github.com/jaypipes/pcidb v1.0.0
       > go: downloading gopkg.in/yaml.v2 v2.4.0
       > go: downloading github.com/mitchellh/go-homedir v1.1.0
       > # github.com/shirou/gopsutil/cpu
       > ../go/pkg/mod/github.com/shirou/gopsutil@v3.20.12+incompatible/cpu/cpu_darwin_cgo.go:13:5: error: 'TARGET_OS_MAC' is not defined, evaluates to 0 [-Werror,-Wundef-prefix=TARGET_OS_]
       > #if TARGET_OS_MAC
       >     ^
       > 1 error generated.
       For full logs, run 'nix log /nix/store/8iqqlbzyk6mflmmzh8cc89h4aynb7r6g-gotop-4.2.0.drv'.
```

### Additional context

Happy to provide more info if needed.

### Notify maintainers

@zowoq 
<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
- system: `"x86_64-darwin"`
 - host os: `Darwin 22.3.0, macOS 10.16`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.13.2`
 - channels(<user_redacted>): `"nixpkgs, nixpkgs-22.11-darwin-22.11-darwin"`
 - channels(root): `""`
 - nixpkgs: `/Users/<user_redacted>/.nix-defexpr/channels/nixpkgs`
```

--- a/pkgs/tools/system/gotop/default.nix
+++ b/pkgs/tools/system/gotop/default.nix
@@ -27,6 +27,9 @@ buildGoModule rec {
 
   ldflags = [ "-s" "-w" "-X main.Version=v${version}" ];
 
+  # prevent `error: 'TARGET_OS_MAC' is not defined`
+  env.CGO_CFLAGS = "-Wno-undef-prefix";
+
   nativeBuildInputs = [ installShellFiles ];
 
   buildInputs = lib.optionals stdenv.isDarwin [ IOKit ];

